biorxiv/article_distances/biorxiv_corrected_published_rate.py

```diff
@@ -0,0 +1,520 @@
+                rate=lambda x: x.published/x.posted
+            )
+        )
+    )
+    publish_rate_df.sample(10, random_state=100)
+
+
+# In[22]:
+
+
+if False:
+    est_published = (
+        publish_rate_df
+        .query("label=='2020 Estimate'")
+        .published
+        .sum()
+    )
+
+    est_posted = (
+        publish_rate_df
+        .query("label=='2020 Estimate'")
+        .posted
+        .sum()
+    )
+
+
+# # Plot Publication Rate
+
+# In[23]:
+
+
+color_mapper = {
+    "2018":"#a6cee3",
+    "2020ML":"#33a02c",
+    "2020":"#1f78b4",
+}
+
+
+# In[24]:
+
+
+g = (
+    p9.ggplot(
+        publish_rate_df
+        .rename(index=str, columns={"label":"Label"})
+    )
+    + p9.aes(
+        x="pub_month", y="rate", 
+        fill="Label", group="Label", 
+        color="Label", linetype="Label",
+        shape="Label"
+    )
+    + p9.geom_point(
+        size=2
+    )
+    + p9.geom_line()
+    + p9.scale_linetype_manual(
+        ['solid', 'solid', 'solid']
+    )
+    + p9.theme_seaborn(
+        style='ticks', 
+        context='paper',
+        font_scale=1.3
+    )
+    + p9.theme(
+        figure_size=(10,4.5),
+        axis_text_x=p9.element_blank(),
+        axis_title_x=p9.element_text(
+            weight='bold',
+            margin={"t":15}
+        ),
+        axis_title_y=p9.element_text(weight='bold'),
+    )
+    + p9.labs(
+        y="Proportion Published",
+        x="Month"
+    )
+    + p9.scale_color_manual(
+        [
+            color_mapper['2020'],
+            color_mapper['2020ML'],
+            color_mapper['2018']
+        ]
+    )
+    + p9.scale_fill_manual(
+        [
+            color_mapper['2020'],
+            color_mapper['2020ML'],
+            color_mapper['2018']
+        ]
+    )
+    + p9.scale_shape_manual(
+        ['o', 'o', 'o']
+    )
+    
+    # plot the x axis titles
+    + p9.geom_vline(xintercept=[2.5, 14.5, 26.5, 38.5, 50.5, 62.5, 74.5])
+    + p9.geom_text(label="2014", x=8.5, y=0, color="black")
+    + p9.geom_text(label="2015", x=20.5, y=0, color="black")
+    + p9.geom_text(label="2016", x=32.5, y=0, color="black")
+    + p9.geom_text(label="2017", x=44.5, y=0, color="black")
+    + p9.geom_text(label="2018", x=56.5, y=0, color="black")
+    + p9.geom_text(label="2019", x=68.5, y=0, color="black")
+    
+    # Plot the overall proportion published
+    + p9.geom_hline(
+        yintercept=0.4196, 
+        linetype='solid', 
+        color=color_mapper['2018']
+    )
+    
+    + p9.geom_hline(
+        yintercept=published/posted, 
+        linetype="solid", 
+        color=color_mapper['2020ML']
+    )
+    
+    + p9.annotate(
+        "text", x=8.5, y=0.395, 
+        label="overall: 0.4196", size = 8
+    )
+    
+    + p9.annotate(
+        "text", x=8.5, y=0.48, 
+        label=f"overall: {published/posted:.4f}", size = 8
+    )
+)
+g.save("output/figures/publication_rate.svg", dpi=500)
+g.save("output/figures/publication_rate.png", dpi=500)
+print(g)
```

Comment:
This figure is super cool! Based on the way that both curves tail off it looks like after two years papers are pretty much published if they ever will be.

Also I feel like the overall proportion published lines are kind of hard to interpret. I read them as "because there have been more preprints over time the proportion published in 2020 is higher than in 2018 since ~75 percent get published regardless." I know that's not necessarily true though because each point on the chart corresponds to a different number of preprints/publications, which could skew the lines up or down as well.

Reply:
Agree that this plot is very cool to see! I am also a little confused by  some of the trends...so my take on this plot is that you have proportion of preprints that get published on your y-axis and time on the x. When the curve trailed off, I initially thought that that mean that many recent preprint were not getting published. However, I realized that this trailing off is likely because these preprints are still in review. I'm not sure how to best account...not sure if "in review" is available as metadata anywhere.

I'm not following Ben's point but perhaps your comment will clarify things

Last, when you calculate missing links. Are these preprints that you imputed were published from similarity distance to a published paper, or are these the preprints where you manually found the actual published paper where the link was missing?

Reply:
Nice fig!
I'm trying to mesh your results with the figure I made:
![image](https://user-images.githubusercontent.com/56938492/90543964-6a96ed80-e154-11ea-9c63-c8d3c7ae4bac.png)
This is theoretically a non parametric method and I've plotted error bars (they're just tiny for all of biorxiv). X axis is time since first shared on biorxiv.
I'm not sure why my overall publication rate settles to such a different value than your figure. This is using data that is pretty equivalent to your crossref mapping I believe. I haven't had a chance to look at your new embedding based method, but your mapping makes it look like it's not that big of a difference?
I haven't been particularly careful about versions or retractions. Do you do anything regarding these that might account for the differences?

Reply:
oh, for figure aesthetics/to address some of the other comments. Perhaps vertical lines to represent error bars assuming a bionomial process might give a sense of publication volume over time and which deviations are significant... But this could also dilute the main point of the figure so ¯\_(ツ)_/¯

Reply:
> so my take on this plot is that you have proportion of preprints that get published on your y-axis and time on the x.

You are correct.

> When the curve trailed off, I initially thought that that mean that many recent preprint were not getting published. However, I realized that this trailing off is likely because these preprints are still in review. I'm not sure how to best account...not sure if "in review" is available as metadata anywhere.

There's no way to account for that unless the authors post an open review. I do not know the stats for the number of open reviews; however, I don't think it is that many. 

>"because there have been more preprints over time the proportion published in 2020 is higher than in 2018 since ~75 percent get published regardless." I know that's not necessarily true though because each point on the chart corresponds to a different number of preprints/publications, which could skew the lines up or down as well.

>I'm not following Ben's point but perhaps your comment will clarify things

Basically, Ben is saying that the overall number of preprints being published is higher in 2020 than in 2018. This is true as enough time has past for preprints to be published; however, the catch here is we don't know the true proportion  of preprints being published as papers are published behind paywalls etc. So trying to compare the overall proportion published (horizontal lines), is not easy as there are too many co-founders to 100% agree with the gap.

>Last, when you calculate missing links. Are these preprints that you imputed were published from similarity distance to a published paper, or are these the preprints where you manually found the actual published paper where the link was missing?

The links were imputed via my similarity distance metric and then classified after I manually checked to see if bioRxiv already contained the published link.

Reply:
> I'm not sure why my overall publication rate settles to such a different value than your figure. This is using data that is pretty equivalent to your crossref mapping I believe. I haven't had a chance to look at your new embedding based method, but your mapping makes it look like it's not that big of a difference?I haven't been particularly careful about versions or retractions. Do you do anything regarding these that might account for the differences?

The reason ours looks different is that my estimate includes my imputed missing links. This pushes the overall proportion up by about 5%. If you take a look at the 2020 Snapshot curve it should be similar to what you are plotting.

>Perhaps vertical lines to represent error bars assuming a bionomial process might give a sense of publication volume over time and which deviations are significant... But this could also dilute the main point of the figure so ¯_(ツ)_/¯

Thanks for the suggestion. Will make a new PR if this pans out.